libs/types-common/src/Data/Id.hs

```diff
@@ -183,6 +186,12 @@ instance FromJSON ClientId where
             fail "Invalid ClientId"
         return (ClientId x)
 
+instance FromJSONKey ClientId where
```

Comment:
How about a little refactor and adding `clientIdFromText :: Text -> Parser (Id a)` to avoid duplication of code in `FromJSONKey` and `FromByteString` ?